evdns.c

```diff
@@ -3643,6 +3645,13 @@ evdns_get_default_hosts_filename(void)
 	char path[MAX_PATH+1];
 	static const char hostfile[] = "\\drivers\\etc\\hosts";
 	char *path_out;
+#if UWP

```

Comment:
Is it defined somewhere? Anyway it is too short, verbose is good in this case, let's use the `WINAPI_FAMILY_PARTITION(WINAPI_PARTITION_DESKTOP)`

Reply:
SHGetSpecialFolderPathA is not available in UWP, per its documentation.

I placed the endif too late, oops, right.

kernel32.dll is not loaded dynamically here as you describe.

You want both GetSystemDirectory and GetWindowDirectory + system32?

Reply:
>You want both GetSystemDirectory and GetWindowDirectory + system32?

Yes, but only if the result may differ

Reply:
>I placed the endif too late, oops, right.

This one does not fixed either